### ISSUE TITLE
A heap with nodes with duplicate values gives an unfixable invalid heap

### ISSUE THREAD

**Issue by k290** [datastructures-js/heap#36]
Heaps should be considered valid when duplicates exist.

According to the definition of binary heaps this is a valid heap:

> All nodes are either [greater than or **equal to**](max heaps) or [less than or **equal to**](min heaps) each of its children, according to a comparison predicate defined for the heap.

Code to reproduce the issue
```
type HeapType = {
  value: number;
};

const minHeap = new MinHeap<HeapType>((x) => x.value);

minHeap.insert({ value: 2268 });
minHeap.insert({ value: 2268 });

console.log(minHeap.fix().isValid());
//false

const minHeap2 = new MinHeap();
minHeap2.insert(22);
minHeap2.insert(22);
console.log(minHeap2.fix().isValid());
//false
```

**Comment by eyas-ranjous** [comment]
@k290 @dawaltconley fix is released v4.3.3

**Comment by k290** [comment]
Thank you very much
